dask/blockwise.py

```diff
@@ -617,7 +617,15 @@ def clone(
 
         indices = []
         for k, idxv in self.indices:
-            if idxv is not None and k in names:
+            try:
+                in_names = k in names
+            except TypeError:
+                # blockwise() where one or more args are sequences of literals;
+                # e.g. k = (list, [0, 1, 2])
+                # See https://github.com/dask/dask/issues/8978
```

Comment:
This is a ugly hack. Clearly, k is not an index in this case. The problem is in `dask.blockwise.blockwise` and `dask.array.blockwise.blockwise`, but properly correcting it in those functions would require putting one's hands in a can of worms that is not annotated and not properly documented. Neither of the functions mentions that \*args can contain literals; however the whole of dask explodes if you add an assertion that the passed \*args match the docstring.